docs/writing-pages.md

```diff
@@ -0,0 +1,186 @@
+1. getAboveTheFoldCount (until we've rendered all the above-the-fold content)
+
+Once we reach the above the fold content, we'll start sending javascript.
+
+If you don't provide a page lifecycle method, we provide a "best-guess" default
+value; for instance, you can omit the head methods altogether and get a
+performant and featureful page.
+
+
+# Examples
+
+### Full page
+
+A `react-server` page that serves a full webpage.
+
+```js
+import HttpStatus from 'http-status-codes';
+import MobileEnabled from ('./middleware/MobileEnabled');
+const ExampleComponent = require("./components/example-component");
+const ExampleStore = require("./stores/example-store");
+const exampleAction = require("./actions/example-action");
+
+class ExamplePage {
+	// See [writing middleware](/docs/writing-middleware) for how to write middleware
+	static middleware() { return [MobileEnabled]; }
+
+	handleRoute() {
+		var params = this.getRequest().getQuery();
+		this._exampleStore = new ExampleStore({
+			id: +params.id
+		});
+	}
+
+	getTitle() {
+		return "Example page"
+	}
+
+	getHeadStylesheets() {
+		return [
+			"/styles/example.css",
+			"/styles/reset.css"
+		]
+	}
+
+	getMetaTags() {
+		var tags = [
+			{ name: "example", content: "Demonstrate a full react-server page" },
+		];
+		return tags;
+	}
+
+	getLinkTags() {
+		return [
+			// prefetch analytics to improve performance
+			{ rel: "prefetch", href: "//www.google-analytics.com" },
+		];
+	}
+
+	getBodyClasses() {
+		return ["responsive-page", "typography"];
+	}
+
+	getElements() {
+		return [
+			<ExampleComponent handleOnClick={exampleAction} {...this._exampleStore} />
+		];
+	}
+}
+```
+
+### Json endpoint
+
+```js
+// returns a promise for example data
+const getExampleData = require("./helpers/get-example-data");
+
+module.exports = class ExampleJsonPage {
+
+	// see the example in [writing middleware](/docs/writing-middleware)
+	static middleware() { return [JsonEndpoint] }
+
+	handleRoute() {
+		const id = this.getRequest().getRouteParams().id;
+		this.data = getExampleData(id);
+		return {code:200};
+	}
+
+	getResponseData() {
+		return this.data;
+	}
+}
+```
+
+### Setting Config values
+
+For instance, to make a page into a fragment by setting the `isFragment` config
+value
+
+```js
+const exampleComponent = require("./components/example-component");
+const exampleStore = require("./stores/example-store");
+
+export default class ExampleFragmentPage {
+	setConfigValues() { return { isFragment: true }; }
+
+	handleRoute() {
+		this._store = exampleStore(this._getStoreOpts());
+		return {code:200};
+	}
+
+	getTitle () {
+		return "Corvair School Fragment";
+	}
+
+	getElements() {
+		return [
+			RootElements.createRootElementWhenResolved(this._store, <exampleComponent/>),
+		];
+	}
+
+	_getStoreOpts() {
+		var storeOpts = {};
+		var routeParams = this.getRequest().getRouteParams();
+		if (routeParams) {
+			storeOpts.exampleId = routeParams.exampleId;
+		}
+		return storeOpts;
+	}
```

Comment:
This `_getStoreOpts` method seems superfluous.  Doesn't really contribute to understanding of `react-server`.  Simpler's better.